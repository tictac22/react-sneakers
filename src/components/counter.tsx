import { useState } from "react"

export const Counter = () => {
	const [counter, setCounter] = useState(0)
	return (
		<div>
			<button onClick={() => setCounter((p) => p + 1)}>increment</button>
			<p>Counter: {counter}</p>
			<button onClick={() => setCounter((p) => p - 1)}>decrement</button>
		</div>
	)
}
