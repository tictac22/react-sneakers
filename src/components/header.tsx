import styles from "../styles/header.module.scss";
export const Header:React.FC = () => {
    return(
        <header className={styles.header}>
            <div className="container">
                <div className={styles.header__container}>
                    <div className={styles.logo}>
                        <div className={styles.logo__img}>
                            <img src="/images/logo.jpg" alt="logo"/>
                        </div>
                        <div className={styles.logo__text}>
                            <h2 className={styles.logo__title}>
                                REACT SNEAKERS
                            </h2>
                            <p className={styles.subtitle}>Магазин лучших кроссовок</p>
                        </div>
                    </div>
                    <nav className={styles.nav}>
                        <div className={styles.nav__cart}>
                            <img src="/svg/cart.svg" alt="cart"/>
                            <p className={styles.nav__price}><span>1205</span>руб.</p>
                        </div>
                        <div className={styles.nav__favorites}>
                            <img src="/svg/favorite.svg" alt="favorite"/>
                        </div>
                        <div className={styles.nav__profile}>
                            <img src="/svg/profile.svg" alt="profile"/>
                        </div>
                    </nav>
                </div>
            </div>
        </header>
    )
}