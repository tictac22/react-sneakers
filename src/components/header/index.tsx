
import { useAppSelector ,useAppDispatch} from "../../redux/hooks";
import { checkMobileMenu } from "../../redux/menu";
import styles from "../../styles/header.module.scss";
import { NavBar } from "./navbar";
export const Header:React.FC = () => {
    const { showMobileMenu} = useAppSelector(state=>state.menu);
    const dispatch = useAppDispatch();
    const toShowMobileMenu = ():void => {
        dispatch(checkMobileMenu(true))
    }
    const hideMobileMenu = ():void => {
        dispatch(checkMobileMenu(false))
    }
    return(
        <header className={styles.header}>
            <div onClick={hideMobileMenu} className={`${styles.header__shadow} ${showMobileMenu ? styles.header__shadow__active : ""}`}></div>
            <div className="container">
                <div className={styles.header__container}>
                    <div className={styles.logo}>
                        <div className={styles.logo__img}>
                            <img src="/images/logo.jpg" alt="logo"/>
                        </div>
                        <div className={styles.logo__text}>
                            <h2 className={styles.logo__title}>
                                REACT SNEAKERS
                            </h2>
                            <p className={styles.logo__subtitle}>Магазин лучших кроссовок</p>
                        </div>
                    </div>
                    <nav className={styles.nav}>
                        <div onClick={toShowMobileMenu} className={styles.nav__burger}>
                            <span></span>
                        </div>
                        <NavBar mobile={showMobileMenu}/>
                    </nav>
                </div>
            </div>
        </header>
    )
}