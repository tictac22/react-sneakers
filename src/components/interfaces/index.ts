export interface ICartItem {
	title: string
	price: number
	img: string
	id: string
}
