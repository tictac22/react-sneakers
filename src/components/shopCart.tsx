import { useAppDispatch, useAppSelector } from "../redux/hooks";
import styles from "../styles/shopcart.module.scss";
import {checkMenu} from "../redux/menu";
export const ShopCart:React.FC = () => {
    const {showMenu} = useAppSelector((state=>state.menu));
    const dispatch = useAppDispatch()
    
    const hideMenu = ():void => {
        dispatch(checkMenu(false))
    }
    return (
        <div className={`${styles.cart} ${showMenu? styles.cart__active : "" } `} >
            <div onClick={hideMenu} className={styles.cart__shadow}></div>
            <div className={styles.cart__inner}>
                <h2 className={styles.cart__title}>
                    Корзина
                </h2>
                <div className={styles.cart__grid}>
                    <div className={styles.cartItem}>
                        <div className={styles.cartItem__img}>
                            <img src="/images/shoes/img1.jpg" alt="shoes"/>
                        </div>
                        <div className={styles.cartItem__text}>
                            <h3 className={styles.cartItem__title}>
                                Мужские Кроссовки Nike Air Max 270
                            </h3>
                            <p className={styles.cartItem__subtitle}>
                                12 999 руб.
                            </p>
                        </div>
                        <div className={styles.cartItem__delete}>
                            <svg width="32" height="32" viewBox="0 0 32 32" fill="none" xmlns="http://www.w3.org/2000/svg">
                            <rect x="0.5" y="0.5" width="31" height="31" rx="7.5" fill="white" stroke="#DBDBDB"/>
                            <path d="M20.0799 18.6155L17.6311 16.1667L20.0798 13.718C21.0241 12.7738 19.5596 11.3093 18.6154 12.2536L16.1667 14.7023L13.7179 12.2535C12.7738 11.3095 11.3095 12.7738 12.2535 13.7179L14.7023 16.1667L12.2536 18.6154C11.3093 19.5596 12.7738 21.0241 13.718 20.0798L16.1667 17.6311L18.6155 20.0799C19.5597 21.0241 21.0241 19.5597 20.0799 18.6155Z" fill="#B5B5B5"/>
                            </svg>

                        </div>
                    </div>
                </div>
                <div className={styles.price}>
                    <div className={styles.price__type}>
                        <p className={styles.price__name}>Цена всех товаров</p>
                        <div className={styles.price__dash}></div>
                        <div className={styles.price__money}><span>21 498 </span>руб.</div>
                    </div>
                    <div className={styles.price__type}>
                        <p className={styles.price__name}>Налог 5%: </p>
                        <div className={styles.price__dash}></div>
                        <div className={styles.price__money}><span>21 498 </span>руб.</div>
                    </div>
                    <div className={styles.price__type}>
                        <p className={styles.price__name}>Итого: </p>
                        <div className={styles.price__dash}></div>
                        <div className={styles.price__money}><span>21 498 </span>руб.</div>
                    </div>
                </div>
                <a href="*" className={styles.btn}>Оформить заказ <img src="/svg/arrow.svg"/></a>
            </div>
        </div>
    )
}