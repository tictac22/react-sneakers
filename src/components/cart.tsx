import { useState } from "react";
import { useAppDispatch } from "../redux/hooks";
import { addToCart, removeFromCart } from "../redux/slicers/shop";
import styles from "../styles/cart.module.scss";
import { ICartItem } from "./interfaces";


export const Cart:React.FC<ICartItem> = ({title,price,imgUrl,id}) => {
    const [isAdded,setAdded] = useState<boolean>(false);
    const [isFavorite,setFavorite] = useState<boolean>(false);
    const dispatch = useAppDispatch();
    const addTo = ():void => {
      if(!isAdded) {
        setAdded(!isAdded);
        dispatch(addToCart({id,price}))
      } else {
        setAdded(!isAdded);
        dispatch(removeFromCart({id,price}))
      }
    }
    const favorite = ():void => {
        setFavorite(!isFavorite);
    }
    return (
        <div className={styles.cart}>
            <div className={styles.cart__inner}>
                <div onClick={favorite} className={styles.cart__favorite}>
                    {isFavorite ? 
                        <img alt="favorite" src="/images/favorite.jpg"/> :
                        <img alt="unfavorite" src="/images/unfavorite.jpg"/>
                    }
                </div>
                <div className={styles.cart__img}>
                    <img alt="shoes" src={`/images/shoes/${imgUrl}.jpg`}/>
                </div>
                <h3 className={styles.cart__title}>
                {title}
                </h3>
                <div className={styles.cart__bottom}>
                    <div className={styles.price}>
                        <div className={styles.price__text}>Цена:</div>
                        <div className={styles.price__money}><span>{price}</span> руб.</div>
                    </div>
                    <div onClick={addTo} className={styles.price__add}>
                    {isAdded ? 
                        <img alt="added" src="/images/added.jpg"/> :
                        <img alt="plus" src="/svg/plus.svg"/>
                    }
                        
                    </div>
                </div>
            </div>
        </div>
    )
}

/*
[
  {
    "title": "Мужские Кроссовки Nike Blazer Mid Suede",
    "price": 12999,
    "imgUrl": "img1",
    "id": "3YuQN5SNO0ioX6adbAk3eQ"
  },
  {
    "title": "Мужские Кроссовки Nike Air Max 270",
    "price": 15600,
    "imgUrl": "img2",
    "id": "jRFlfF6IukWair2Jl_z"
  },
  {
    "title": "Мужские Кроссовки Nike Blazer Mid Suede",
    "price": 8499,
    "imgUrl": "img3",
    "id": "Mh5jEbyxKUm80h_z8iVE3Q"
  },
  {
    "title": "Кроссовки Puma X Aka Boku Future Rider",
    "price": 8999,
    "imgUrl": "img4",
    "id": "HxiOc8XOBEWVSDZ7JIaeIA"
  },
  {
    "title": "Кроссовки Future Rider",
    "price": 16999,
    "imgUrl": "img5",
    "id": "dfMWbmPS90Ctp45m9Zslxw"
  },
  {
    "title": "Кроссовки Black Edition",
    "price": 16999,
    "imgUrl": "img6",
    "id": "aj1eFTpaLUCrC5tAJHjOKA"
  },
  {
    "title": "Кроссовки Orange Boomb Edition",
    "price": 7499,
    "imgUrl": "img7",
    "id": "EOEn6hRxC0KGud1Ml1hOhA"
  }
]*/ 