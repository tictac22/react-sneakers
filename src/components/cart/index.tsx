import styles from "../../styles/cart.module.scss"
import { ICartItem } from "../interfaces"
import { AddTo } from "./addTo"
import { Favorite } from "./favorite"

export const Cart: React.FC<ICartItem> = (props) => {
	return (
		<div className={styles.cart}>
			<div className={styles.cart__inner}>
				<Favorite {...props} />
				<div className={styles.cart__img}>
					<img alt="shoes" src={`/images/shoes/${props.img}.jpg`} />
				</div>
				<h3 className={styles.cart__title}>{props.title}</h3>
				<div className={styles.cart__bottom}>
					<div className={styles.price}>
						<div className={styles.price__text}>Cost</div>
						<div className={styles.price__money}>
							<span>${props.price}</span>
						</div>
					</div>
					<AddTo {...props} />
				</div>
			</div>
		</div>
	)
}
