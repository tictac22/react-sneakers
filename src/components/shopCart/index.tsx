import { useAppDispatch, useAppSelector } from "../../redux/hooks";
import styles from "../../styles/shopcart.module.scss";
import {checkMenu} from "../../redux/slicers/menu";
import { ShopCartItem } from "./shopCartItem";
import { useGetAllItems } from "../hooks/getItems";
import {ICartItem} from "../interfaces/"
import { EmptyCart } from "./emptyCart";
import { emptyCart, successCart } from "./textCartItem";
import { useState } from "react";
export const ShopCart:React.FC = () => {
    const {isError,isLoading,cartItems} = useGetAllItems();
    const {showMenu} = useAppSelector((state=>state.menu));
    const {shopItems,totalPrice} = useAppSelector(state=>state.shop);
    const dispatch = useAppDispatch()
    const [isSuccessOrder,setIsSucessOrder] = useState<boolean>(false);
    const [successOrder,showSucessOrder] = useState<boolean>(false);
    const hideMenu = ():void => {
        dispatch(checkMenu(false))
    }
    const discount:number = parseInt((totalPrice / 100 * 5).toFixed(0),10);
    const filteredShopItems = ():ICartItem[] => {
        const filteredArray:ICartItem[] = [];
        cartItems.forEach(item=>{
            shopItems.forEach(number=>{
            if(item.id===number) {
              filteredArray.push(item);
            }
          })
        })
        return filteredArray
    }
    const makeOrder = ():void => {
        setIsSucessOrder(!isSuccessOrder);
        setTimeout(()=>{
            showSucessOrder(!successOrder);
            setIsSucessOrder(false);
        },3000)
    }
    return (
        <div className={`${styles.cart} ${showMenu? styles.cart__active : "" } `} >
            <div onClick={hideMenu} className={styles.cart__shadow}></div>
                <div className={styles.cart__inner}>
                    <h2 className={styles.cart__title}>
                        Корзина
                    </h2>
                    <div className={styles.cart__grid}>
                                {
                                    isLoading ? <p>Loading</p> :
                                    isError ? <p>Error</p> :
                                    successOrder ? <EmptyCart {...successCart}/> :
                                    filteredShopItems().length === 0 ? <EmptyCart {...emptyCart}/> :
                                    filteredShopItems().map(item=><ShopCartItem
                                    id={item.id} 
                                    key={item.id} 
                                    title={item.title} 
                                    price={item.price} 
                                    imgUrl={item.imgUrl}/>)
                                }           
                    </div>
                    { successOrder ? "" :  filteredShopItems().length !== 0 ?  (
                        <>
                            <div className={styles.price}>
                                <div className={styles.price__type}>
                                    <p className={styles.price__name}>Цена всех товаров</p>
                                    <div className={styles.price__dash}></div>
                                    <div className={styles.price__money}><span>{totalPrice} </span>руб.</div>
                                </div>
                                <div className={styles.price__type}>
                                    <p className={styles.price__name}>Налог 5%: </p>
                                    <div className={styles.price__dash}></div>
                                    <div className={styles.price__money}><span>{discount} </span>руб.</div>
                                </div>
                                <div className={styles.price__type}>
                                    <p className={styles.price__name}>Итого: </p>
                                    <div className={styles.price__dash}></div>
                                    <div className={styles.price__money}><span>{totalPrice + discount}</span> руб.</div>
                                </div>
                            </div>
                            {isSuccessOrder ? <div className={styles.loader__container}><div className={styles.loader}></div></div> :
                            <button onClick={makeOrder}  className={`${styles.price__btn} btn`}>
                                <p>Оформить заказ</p>
                                <img src="/svg/arrow.svg"/>
                            </button>
                            }
                        </>
                    ) : ""}
            </div>
        </div>
    )
}