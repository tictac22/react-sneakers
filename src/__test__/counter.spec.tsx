import { fireEvent, render, RenderResult, screen } from "@testing-library/react" // (or /dom, /vue, ...)
import { Counter } from "../components/counter"

describe("counter", () => {
	let component: RenderResult
	beforeEach(() => {
		component = render(<Counter />)
	})
	it("mounts", () => {
		expect(screen.getByText("Counter: 0")).not.toBeNull()
		expect(component.asFragment()).toMatchSnapshot()
	})

	it("increments", () => {
		const incrementButton = screen.getByText("increment")

		fireEvent.click(incrementButton)

		expect(screen.getByText("Counter: 1")).toBeInTheDocument()
	})

	it("decrements", () => {
		const decrementButton = screen.getByText("decrement")

		fireEvent.click(decrementButton)

		expect(screen.getByText("Counter: -1")).toBeInTheDocument()
	})
})
