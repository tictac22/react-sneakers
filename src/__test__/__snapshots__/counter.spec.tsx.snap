// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter mounts 1`] = `
<DocumentFragment>
  <div>
    <button>
      increment
    </button>
    <p>
      Counter: 0
    </p>
    <button>
      decrement
    </button>
  </div>
</DocumentFragment>
`;
